packages/core/src/payment/strategies/paypal-commerce/paypal-commerce-sdk.ts

```diff
@@ -59,13 +60,114 @@ export interface PaypalButtonStyleOptions {
 export interface ButtonsOptions {
     style?: PaypalButtonStyleOptions;
     fundingSource?: string;
-    createOrder?(): Promise<string | void>; // TODO: this method should return only Promise<void>
-    onApprove?(data: ApproveDataOptions): void;
+    createOrder?(): Promise<string>;
+    onApprove?(data: ApproveDataOptions, actions: ApproveActions): void;
+    onShippingChange?(data: ShippingOptionsPayload, actions: ApproveActions, cart: Cart): void;
```

Comment:
As far as I see, you are passing only data to `onShippingChange` in paypal commerce button strategy. It might be better to update the interface.  And we can remove Cart interface import as well.

Reply:
removed